lib/__tests__/input-type.test.ts

```diff
@@ -0,0 +1,70 @@
+import { Factory, HookFn, register } from 'fishery';
+import { factories } from './sample-app/factories';
+
+describe('Input types', () => {
+  interface User {
+    firstName: string;
+    lastName: string;
+    address: { city: string; state: string; country: string };
+  }
+
+  interface UserTransientParams {
+    city: string;
+    state: string;
+    country: string;
+    name: string;
+  }
+
+  const userFactory = Factory.define<User, any, UserTransientParams>(
+    ({ transientParams }) => {
+      const {
+        name = 'Sharon Jones',
+        city = 'Grand Rapids',
+        state = 'MI',
+        country = 'USA',
+      } = transientParams;
+
+      const [firstName, lastName] = name.split(' ');
+
+      return {
+        firstName,
+        lastName,
+        address: {
+          city,
+          state,
+          country,
+        },
+      };
+    },
+  );
+
+  const factories = register({
+    user: userFactory,
+  });
+
+  it('uses default when no transient param passed', () => {
+    const user = factories.user.build();
+    expect(user.address.country).toEqual('USA');
+  });
+
+  it('allows specifying different input type than output type', () => {
```

Comment:
Is "input type" and "output type" left over from the initial `UserInput` in, `User` out approach you mentioned initially exploring?

Reply:
Yes, you're right. I was still using that verbiage in several places. Updated in https://github.com/thoughtbot/fishery/pull/10/commits/4da326cbf0e73ef1dd28f97e3cf75a42748a2804.